### PR TITLE
Fix OpenSSL package source

### PR DESCRIPTION
**Pull request by jstabenow**
openssl.org/source/ -> openssl.org/source/old/1.1.1/

### CODE REVIEW

File: alpine/Dockerfile [hunk 1]
```diff
@@ -97,7 +97,7 @@ RUN apk add --no-cache --virtual .build-deps \
     && cd /tmp \
     && if [ -n "${RESTY_EVAL_PRE_CONFIGURE}" ]; then eval $(echo ${RESTY_EVAL_PRE_CONFIGURE}); fi \
     && cd /tmp \
-    && curl -fSL https://www.openssl.org/source/openssl-${RESTY_OPENSSL_VERSION}.tar.gz -o openssl-${RESTY_OPENSSL_VERSION}.tar.gz \
+    && curl -fSL https://www.openssl.org/source/old/1.1.1/openssl-${RESTY_OPENSSL_VERSION}.tar.gz -o openssl-${RESTY_OPENSSL_VERSION}.tar.gz \
```
Comment (mrobinet) [inline alpine/Dockerfile:100]: Hard-coding `1.1.1` here breaks things if `RESTY_OPENSSL_VERSION` is set to an older version. The path should be derived from the build arg. Similar issues exist with the patching of openssl done below, but that is outside the scope of this PR.

Another problem is that the latest openssl version (currently 1.1.1f) can't be referenced as that is still hosted directly under https://openssl.org/source. Maybe just expose a build arg for the openssl path with a default of https://www.openssl.org/source/old/?

```suggestion
    && curl -fSL https://www.openssl.org/source/old/$(echo $RESTY_OPENSSL_VERSION | sed 's/[a-z]$//')/openssl-${RESTY_OPENSSL_VERSION}.tar.gz -o openssl-${RESTY_OPENSSL_VERSION}.tar.gz \
```

**Comment by neomantra** [comment]
I’m have patch in progress that exposes the URLs and the patch version as build args. Should be out later tonight.

**Comment by neomantra** [comment]
Thanks for working on this and reviewing it.  I just pushed commit ebc648002a1d2dcb20b3e04bc55468bd3eb92ed7 which updates all the build-from-source versions.    

It's building now... I'm doing some work to ensure that the upstream-packaged images are updated with the latest OpenSSL and then will tag this `1.15.8.3-1`

**Comment by jstabenow** [comment]
Yipp, thanks for fast processing and repair

**Comment by neomantra** [comment]
The automated build picked up the latest OpenSSL for the upstream packaged, so I just tagged `1.15.8.3-1`.

**Pull request closed without merging**